src/sugar3/graphics/progressicon.py

```diff
@@ -37,49 +51,58 @@ def __init__(self, icon_name, pixel_size, stroke_color, fill_color,
                  direction='vertical'):
         Gtk.DrawingArea.__init__(self)
 
```

Comment:
Explain the arguments for this function (and every other is this module) and what the function returns.